clevertap-core/src/main/java/com/clevertap/android/sdk/CleverTapAPI.java

```diff
@@ -2890,20 +2890,13 @@ public void setOptOut(final boolean userOptOut, final boolean allowSystemEvents)
             optOutMap.put(Constants.CLEVERTAP_OPTOUT, userOptOut);
```

Comment:
@CTLalit You've mentioned this in the method javadoc -
```
The case where `userOptOut = false` and `allowSystemEvents = false` is invalid.
     * If this combination is provided, the SDK will default to the "Full Opt-In" behavior
```

But this is not added in the code. Your logic works right now, but in the future, someone might miss it or miss the `!` or add a `!` which could lead to potential issues.. I think we should add something specific to the code like this -
```
// Handle invalid combination as documented
if (!userOptOut && !allowSystemEvents) {
    allowSystemEvents = true; // Default to full opt-in
}
```